### DISCUSSION TITLE
Rules

### DISCUSSION THREAD

**Discussion by arielBlum1** [cdisc-org/cdisc-rules-engine#432] [General]
Hi, following our meeting some points we raised about the rules:

-  Missing sensitivity for Variable In example in this rule : CORE-000019 we think the Variable sensitivity is more appropriate 

- Handling missing Variables: How are we supposed to handle the case where the variable is not found, should we add to the rule the check that the variable is found?  For example, in cases where the operator is “empty” ( CG0666 ) and a variable is not found should it get true or false ? (For example in this rule CG0217 they did check it was found before checking other things)

Thanks in advance,
Ariel Blum (Bioforum)

**Comment by apalmercdisc** [comment]
For CORE-000019, the rule editor does not have a sensitivity of "Variable", instead using the rule type "Variable Metadata Check" to build a dataset and then "Record" sensitivity to check each variable in that constructed dataset. It was previously "Dataset" sensitivity and has now been corrected to "Record" sensitivity to account for multiple variables within a single dataset that may have labels that are >40 characters.

**Comment by gerrycampion** [comment]
Handling missing variables: While it wouldn't hurt to include a `not_exists` check for CG0666, it is not necessary because this is checked by CG0016 (if "Variable Core Status = Expected", then "Variable present in dataset"). 
The Rule for CG0666 says: "AGE ^= null". In contrast, the Condition for CG0217 says: "TAETORD present in dataset..." and the Rule says: "TAETORD = SE.TAETORD". Because CG0217 explicitly states a presence check for TAETORD, it is included in the CORE rule logic.  SV.TAETORD is not an ig-modeled variable; therefore it does not have a core (Expected) designation, so it won't be covered by CG0016.

Having the additional presence check for CG0666 would also mean that 2 issues would appear in the report instead of one (CG0016 and CG0666), which is not ideal.
